python/ql/src/experimental/Security/CWE-022/ZipSlip.ql

```diff
@@ -0,0 +1,24 @@
+/**
+ * @name Arbitrary file write during archive extraction ("Zip Slip")
+ * @description Extracting files from a malicious archive without validating that the
+ *              destination file path is within the destination directory can cause files outside
+ *              the destination directory to be overwritten.
+ * @kind path-problem
+ * @id py/zipslip
+ * @problem.severity error
+ * @security-severity 7.5
+ * @precision high
+ * @tags security
+ *       external/cwe/cwe-022
+ */
+
+import python
+import experimental.semmle.python.security.ZipSlip
+import DataFlow::PathGraph
+
+
```

Comment:
This file is not autoformatted. Again, please ensure that all `.ql` and `.qll` files have been autoformatted, as this is a prerequisite for merging.